code/modules/xenoarchaeology/traits/xenoartifact_majors.dm

```diff
@@ -0,0 +1,589 @@
+		victim.electrocute_act(damage, X, 1, 1)
+		playsound(get_turf(X), 'sound/machines/defib_zap.ogg', 50, TRUE)
+	X.cooldownmod = (X.charge*0.1) SECONDS
+
+///============
+/// Timestop, spawns time-stop effect
+///============
+/datum/xenoartifact_trait/major/timestop
+	desc = "Melted"
+	label_desc = "Melted: The shape is drooling and sluggish. Additionally, light around it seems to invert."
+
+/datum/xenoartifact_trait/major/timestop/on_touch(obj/item/xenoartifact/X, mob/user)
+	to_chat(user, "<span class='notice'>Your hand feels slow while stroking the [X.name].</span>")
+	return TRUE
+
+/datum/xenoartifact_trait/major/timestop/activate(obj/item/xenoartifact/X, atom/target)
+	var/turf/T = (get_turf(X?.loc) || get_turf(target?.loc))
+	if(!T)
+		return
+	new /obj/effect/timestop(T, 2, (X.charge*0.2) SECONDS)
+	X.cooldownmod = (X.charge*0.35) SECONDS
+
+///============
+/// Laser, shoots varying laser based on charge
+///============
+/datum/xenoartifact_trait/major/laser
+	desc = "Barreled"
+	label_desc = "Barreled: The shape resembles the barrel of a gun. It's possible that it might dispense candy."
+	flags = PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/major/laser/activate(obj/item/xenoartifact/X, atom/target, mob/living/user)
+	if(isliving(target) && get_dist(target, X.loc || user) <= 1)
+		var/mob/living/victim = target
+		victim.adjust_fire_stacks(5*(X.charge/X.charge_req))
+		victim.IgniteMob()
+		return
+	var/obj/item/projectile/A
+	switch(X.charge)
+		if(0 to 24)
+			A = new /obj/item/projectile/beam/disabler
+		if(25 to 79)
+			A = new /obj/item/projectile/beam/laser
+		if(80 to 200)
+			A = new /obj/item/ammo_casing/energy/laser/heavy
+	A.preparePixelProjectile(get_turf(target), X)
+	A.fire()
+	playsound(get_turf(src), 'sound/mecha/mech_shield_deflect.ogg', 50, TRUE) 
+
+///============
+/// Corginator, turns the target into a corgi for a short time
+///============
+/datum/xenoartifact_trait/major/corginator ///All of this is stolen from corgium. 
+	desc = "Fuzzy" //Weirdchamp
+	label_desc = "Fuzzy: The shape is hard to discern under all the hair sprouting out from the surface. You swear you've heard it bark before."
+	flags = BLUESPACE_TRAIT
+	///List of all affected targets, used for early qdel
+	var/list/victims = list()
+	///list of targets by corgi
+	var/list/target_keys = list()
+	///list of timers for transform back
+	var/list/timers = list()
+
+/datum/xenoartifact_trait/major/corginator/activate(obj/item/xenoartifact/X, mob/living/target)
+	X.say(pick("Woof!", "Bark!", "Yap!"))
+	if(!(SSzclear.get_free_z_level()))
+		playsound(get_turf(X), 'sound/machines/buzz-sigh.ogg', 50, TRUE)
+		return
+	if(istype(target, /mob/living) && !(istype(target, /mob/living/simple_animal/pet/dog/corgi)))
+		var/mob/living/simple_animal/pet/dog/corgi/new_corgi = transform(target)
+		timers[new_corgi] = addtimer(CALLBACK(src, .proc/transform_back, new_corgi), (X.charge*0.6) SECONDS, TIMER_STOPPABLE)
```

Comment:
Why are you keeping references to the timer IDs if you aren't doing anything with them? Seems like a pointless hard reference to new_corgi that never gets cleared